infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -326,19 +329,20 @@ private void createOutgoingSessionsBySubscriber(ReplicationSubscriber subscriber
                 subscriber, sessionsToAdd);
 
         configManager.generateConfig(sessionsToAdd, true);
+        return sessionsToAdd;
     }
 
-    private void createIncomingSessionsBySubscriber(ReplicationSubscriber subscriber) {
+    public void createIncomingSessionsBySubscriber(ReplicationSubscriber subscriber, LogReplicationSession sessionFromSource) {
```

Comment:
I suggest adding a new method which adds a particular session than overriding the current method.  In both cases, a new session and metadata are added.  It is difficult to understand what difference the check makes.